x/avs/types/genesis.go

```diff
@@ -1,21 +1,163 @@
 package types
 
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	assetstype "github.com/ExocoreNetwork/exocore/x/assets/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
 // this line is used by starport scaffolding # genesis/types/import
 
 // DefaultIndex is the default global index
 const DefaultIndex uint64 = 1
 
-// DefaultGenesis returns the default genesis state
-func DefaultGenesis() *GenesisState {
+func NewGenesisState(
+	avsInfos []AVSInfo,
+	taskInfos []TaskInfo,
+	blsPubKeys []BlsPubKeyInfo,
+	taskResultInfos []TaskResultInfo,
+	challengeInfos []ChallengeInfo,
+	taskNums []TaskID,
+	chainIDInfos []ChainIDInfo,
+) *GenesisState {
+	// Ensure slices are never nil
+	if avsInfos == nil {
+		avsInfos = []AVSInfo{}
+	}
+	if taskInfos == nil {
+		taskInfos = []TaskInfo{}
+	}
+	if blsPubKeys == nil {
+		blsPubKeys = []BlsPubKeyInfo{}
+	}
+	if taskResultInfos == nil {
+		taskResultInfos = []TaskResultInfo{}
+	}
+	if challengeInfos == nil {
+		challengeInfos = []ChallengeInfo{}
+	}
+	if taskNums == nil {
+		taskNums = []TaskID{}
+	}
+	if chainIDInfos == nil {
+		chainIDInfos = []ChainIDInfo{}
+	}
 	return &GenesisState{
-		Params: DefaultParams(),
+		AvsInfos:        avsInfos,
+		TaskInfos:       taskInfos,
+		BlsPubKeys:      blsPubKeys,
+		TaskResultInfos: taskResultInfos,
+		ChallengeInfos:  challengeInfos,
+		TaskNums:        taskNums,
+		ChainIdInfos:    chainIDInfos,
 	}
 }
 
+// DefaultGenesis returns the default genesis state
+func DefaultGenesis() *GenesisState {
+	return NewGenesisState(nil, nil, nil, nil, nil, nil, nil)
+}
+
 // Validate performs basic genesis state validation returning an error upon any
 // failure.
 func (gs GenesisState) Validate() error {
-	// this line is used by starport scaffolding # genesis/types/validate
+	// Check for duplicated avs address
+	avsAddresses := make(map[string]bool)
+	for _, info := range gs.AvsInfos {
+		if !common.IsHexAddress(info.AvsAddress) {
```

Comment:
Validate the case of these addresses? Or is it not necessary? Please check

Reply:
do SetXxx functions convert it